src/Jackett.Common/Indexers/FunFile.cs

```diff
@@ -126,14 +116,17 @@ protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuer
                     var qTimeAgo = row.QuerySelector("td:nth-of-type(6)");
                     var qSize = row.QuerySelector("td:nth-of-type(8)");
 
-                    var catStr = qCatLink.GetAttribute("href").Split('=')[1].Split('&')[0];
-                    release.Category = MapTrackerCatToNewznab(catStr);
+                    if (qDownloadLink == null)
+                        throw new Exception("Download link not found. Make sure you can download from the website.");
 
                     release.Link = new Uri(SiteLink + qDownloadLink.GetAttribute("href"));
                     release.Title = qDetailsLink.GetAttribute("title").Trim();
                     release.Comments = new Uri(SiteLink + qDetailsLink.GetAttribute("href"));
                     release.Guid = release.Link;
 
+                    var catStr = qCatLink.GetAttribute("href").Split('=')[1].Split('&')[0];
+                    release.Category = MapTrackerCatToNewznab(catStr);
```

Comment:
I know I just mentioned it in an already closed PR, but some variation on the code below is less error prone if the url format changes.
```
var qParams = HttpUtility.ParseQueryString(new Uri(qCatLink.GetAttribute("href")).Query);
var catStr = qParams["catParam"];
```

Reply:
If you take a look at the previous lines, we are already hardcoding that cat is the first query param.
`var qCatLink = row.QuerySelector("a[href^=\"browse.php?cat=\"]");`
I didn't change that code, it's not the goal of this PR.